app/main.py

```diff
@@ -1,8 +1,38 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        self.name = name
+        self.age = age
+
+        Person.people[self.name] = self
+
+
+def find_partner(person_obj: Person,
+                 person_to_find: str | None
+                 ) -> Person | str | None:
+    if person_to_find in Person.people:
+        partner = person_obj.people[person_to_find]
```

Comment:
Call `people` class attribute from the `Person` class as it is done on line 21, not from the class instance.